src/controllers/userControllers.ts

```diff
@@ -94,6 +94,17 @@ class UserControllers {
     }
   }
 
+  async getUserPosts(req: Request, res: Response, next: NextFunction) {
+    const { userId } = req.params;
+    try {
+      const posts = await postService.findPostsByUserId(userId, req.user.id);
```

Comment:
Let's add a `userService.findUserById()` here to make sure the user with the id `userId` exists before loading their posts.